### PR TITLE
src/citrine/informatics documentation for orion

### PR DESCRIPTION
**Pull request by kyle-laplante**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.

### CODE REVIEW

File: src/citrine/informatics/constraints.py [hunk 1]
```diff
@@ -24,7 +26,15 @@ def get_type(cls, data):
 
 
 class ScalarRangeConstraint(Serializable['ScalarRangeConstraint'], Constraint):
-    """A Citrine ScalarRangeConstraint."""
+    """Specifies the material attributes that a user wishes to constrain in scalar format.
```
Comment (sparadiso) [inline src/citrine/informatics/constraints.py]:
```suggestion
    """Represents an inequality constraint on a scalar-valued material attribute.
```

File: src/citrine/informatics/design_spaces.py [hunk 2]
```diff
@@ -67,7 +76,15 @@ def __str__(self):
 
 
 class EnumeratedDesignSpace(Resource['EnumeratedDesignSpace'], DesignSpace):
-    """Design space composed of enumerated items."""
+    """Design space composed of enumerated items.
```
Comment (sparadiso) [inline src/citrine/informatics/design_spaces.py]:
```suggestion
    """Design space composed of an explicit enumeration of candidates. Note that every candidate must have exactly the descriptors in the list populated (no more, no less) to be included. 
```

**Comment by sparadiso** [comment]
@kyle-laplante the formatting is slightly different from the pattern Brendan laid out (e.g., https://github.com/CitrineInformatics/citrine-python/blob/develop/src/citrine/resources/material_run.py#L21) but not dramatically.

**Comment by bfolie** [comment]
> @kyle-laplante the formatting is slightly different from the pattern Brendan laid out (e.g., https://github.com/CitrineInformatics/citrine-python/blob/develop/src/citrine/resources/material_run.py#L21) but not dramatically.

I believe that the format needs to follow numpy style (https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html) so that the Sphinx plugin will automatically render it into docstrings. This style might work as well, but you should build the docs locally and check before continuing.

File: src/citrine/informatics/constraints.py [hunk 3]
```diff
@@ -52,7 +68,15 @@ def __str__(self):
...
 class CategoricalConstraint(Serializable['CategoricalConstraint'], Constraint):
-    """A Citrine CategoricalConstraint."""
+    """Specifies the categorical material attributes that a user wishes to constrain.
+
+    Parameters
+    ----------
+    descriptor_key: str
+        the key corresponding to a descriptor
+    acceptable_categories: list[str]
+        the names of the acceptable categories to constrain on
```
Comment (sparadiso) [inline src/citrine/informatics/constraints.py]:
```suggestion
        the names of the acceptable categories to constrain to
```

File: src/citrine/informatics/constraints.py [hunk 4]
```diff
@@ -52,7 +68,15 @@ def __str__(self):
 
 
 class CategoricalConstraint(Serializable['CategoricalConstraint'], Constraint):
-    """A Citrine CategoricalConstraint."""
+    """Specifies the categorical material attributes that a user wishes to constrain.
+
+    Parameters
+    ----------
+    descriptor_key: str
+        the key corresponding to a descriptor
```
Comment (sparadiso) [inline src/citrine/informatics/constraints.py]:
```suggestion
        the key corresponding to the associated Categorical descriptor
```

File: src/citrine/informatics/constraints.py [hunk 5]
```diff
@@ -52,7 +68,15 @@ def __str__(self):
 
 
 class CategoricalConstraint(Serializable['CategoricalConstraint'], Constraint):
-    """A Citrine CategoricalConstraint."""
+    """Specifies the categorical material attributes that a user wishes to constrain.
```
Comment (sparadiso) [inline src/citrine/informatics/constraints.py]:
```suggestion
    """Represents a constraint on a categorical material attribute to be one of a set of acceptable values.
```

File: src/citrine/informatics/descriptors.py [hunk 6]
```diff
@@ -20,7 +22,17 @@ def get_type(cls, data) -> Type[Serializable]:
...
 
 class RealDescriptor(Serializable['RealDescriptor'], Descriptor):
-    """A real descriptor."""
+    """Captures domain-specific context about the bounds of an integer-valued datum.
+
+    Parameters
+    ----------
+    key: str
+        the key corresponding to a descriptor
+    lower_bound: int
```
Comment (sparadiso) [inline src/citrine/informatics/descriptors.py]:
```suggestion
    lower_bound: float
```

File: src/citrine/informatics/descriptors.py [hunk 7]
```diff
@@ -20,7 +22,17 @@ def get_type(cls, data) -> Type[Serializable]:
...
+        inclusive lower bound for valid real values
+    upper_bound: int
```
Comment (sparadiso) [inline src/citrine/informatics/descriptors.py]:
```suggestion
    upper_bound: float
```

File: src/citrine/informatics/descriptors.py [hunk 8]
```diff
@@ -49,7 +61,15 @@ def __init__(self,
...
 class InorganicDescriptor(Serializable['InorganicDescriptor'], Descriptor):
-    """A real descriptor."""
+    """Captures domain-specific context about the chemical formula for an inorganic compound.
+
+    Parameters
+    ----------
+    key: str
+        the key corresponding to a descriptor
+    threshold: float
+        the threshold for valid chemical formulae
```
Comment (sparadiso) [inline src/citrine/informatics/descriptors.py]:
```suggestion
        the threshold for valid chemical formulae. Users can think of this as a level of tolerance for typos and/or loss in interpreting a string input as a parseable chemical formula.
```

File: src/citrine/informatics/design_spaces.py [hunk 9]
```diff
@@ -28,7 +30,17 @@ def get_type(cls, data) -> Type[Serializable]:
 
 
 class ProductDesignSpace(Resource['ProductDesignSpace'], DesignSpace):
-    """Design space composed of an outer product of univariate dimensions."""
+    """Design space composed of an outer product of univariate dimensions (i.e. sets), either continuous or enumerated.
```
Comment (sparadiso) [inline src/citrine/informatics/design_spaces.py]:
```suggestion
    """Design space composed of an outer product of univariate dimensions, either continuous or enumerated.
```

Comment (sparadiso) [reply src/citrine/informatics/design_spaces.py]: I'm not totally sure about the `set` language here.

File: src/citrine/informatics/predictors.py [hunk 10]
```diff
@@ -42,7 +44,23 @@ def get_type(cls, data) -> Type[Serializable]:
 
 
 class SimpleMLPredictor(Serializable['SimplePredictor'], Predictor):
-    """A predictor that predicts an output using a machine-learned model."""
+    """A predictor that predicts an output using a machine-learned model.
```
Comment (sparadiso) [inline src/citrine/informatics/predictors.py]:
```suggestion
    """A simplified predictor interface that builds a graphical model connecting the set of inputs through latent variables to the outputs. Supported complex inputs (such as chemical formulas) are auto-featurized and machine learning models are built for each latent variable and output.
```

**Pull request merged**